### DISCUSSION TITLE
How to work on a project that can be installed on servers without composer possibility?

### DISCUSSION THREAD

**Discussion by Firehold** [composer/composer#12168] [Q&A / Support]
Hi folks,

i'm relatively new to composer. I like the benefits, but I'm also struggling with some usecases.

I am working on a project with some others via github. In this project, composer was used to implement some packages/libraries.
 initially
Normally these packages are not pushed to your own git repo because they should be managed by composer on the final web server, but our project is often installed on small web servers that don't have the ability to install and work with composer.
Since the packages are usually pulled directly from their github pages from composer, I can't push them to our own repo.

What is the recommended way here? I usually implemented the libraries manually before working with composer because of that i dont have expiriences with that.

I hope you can understand what I am struggling with.

Many Greets,
Christian

**Comment by Seldaek** [comment]
That'd be a case where I'd say you should make sure to run `composer install` before, zip up the result as a deliverable archive, and deploy that archive to your servers that have no ability to run Composer. That install and zipping up is typically done in a CI build step, but does not have to be.

**Reply by Firehold** [reply]
Thank you for your answer.
Is github able to run this with or before creating a release?
For example we implemented fontawesome into the project. Unfortunatelly with the git folder inside it has a size of 150 MB, when deleting th .git and .github folder manually its 31,4 MB. When i understood correctly the "composer install" command will implement the package and get rid of these, for productive environment, obsolete files?

**Reply by Seldaek** [reply]
Composer will by default install from zip files which doesn't contain the full .git folder with history yes.
